### PR TITLE
feat: new env permissions && permission tweaks

### PR DESCRIPTION
**Pull request by benstoltz**
1. Description: Adds new env permissions and slight tweaks to projects initiative workspace/tab permissions.

1. Instructions for testing:

1. Closes Issues: #<number> (if appropriate)

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Review by juliannemarik — APPROVED** [review]
These changes look good to me...wasn't there something we wanted to do in the `wellKnownCatalog` module to swap out the "project" collection with the "projectAndInitiative" collection? Or are we not doing that yet?

**Comment by benstoltz** [comment]
We aren't doing that yet, these are still QA gated. Next pass through then we should be doing that swap I believe.

File: packages/common/src/permissions/HubPermissionPolicies.ts [hunk 1]
```diff
@@ -108,6 +108,18 @@ const SystemPermissionPolicies: IPermissionPolicy[] = [
...
+    environments: ["qaext"],
+  },
+  {
+    permission: "hub:environment:devext",
+    environments: ["devext"],
+  },
+  {
+    permission: "hub:environment:production",
+    environments: ["production"],
+  },
```
Comment (jordantsanz) [inline packages/common/src/permissions/HubPermissionPolicies.ts:122]: I'd maybe add a quick comment here saying to use these only if necessary -- we'd rather people make permissions where they specify the environment, but if that can't happen (i.e. in the case of notices where there isn't an entity type to write permissions for), then we can utilize these.

Comment (dbouwman) [reply packages/common/src/permissions/HubPermissionPolicies.ts:122]: I agree with @jordantsanz - these should be limited use, and we definitely don't want to have them used in the dependencies arrays. But I do like using permissions >> `context.environment === "qaext"` sort of stuff

**Review by dbouwman — APPROVED** [review]

File: packages/common/src/permissions/HubPermissionPolicies.ts (hunk 1 above)
Comment (benstoltz) [reply packages/common/src/permissions/HubPermissionPolicies.ts:125]: Added :)

**Pull request merged**